ui/src/app/pages/admin/admin-lock-request.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {
+  SemiBoldHeader
+} from 'app/components/headers';
+import {DatePicker, TextArea} from 'app/components/inputs';
+import {Button} from 'app/components/buttons';
+import {workspaceAdminApi} from 'app/services/swagger-fetch-clients';
+import {useState} from 'react';
+import {TooltipTrigger} from 'app/components/popups';
+import colors from 'app/styles/colors';
+
+interface Props {
+  workspace: string,
+  onLock: Function,
+  onCancel: Function
+}
+
+export const AdminLockRequest = (props: Props) => {
+  const [requestReason, setRequestReason] = useState('');
+  const [requestDate, setRequestDate] = useState(new Date());
+  const [showError, setShowError] = useState(false);
+
+  const enableLockButton = requestReason?.length > 0 && requestDate?.toString() !== ''
+      && !isNaN(requestDate.valueOf()) && !showError;
+
+
+  const getToolTipContent = showError ? 'Error occurred while Locking Workspace' :
+      'Request Reason & Valid Request Date (in YYYY-MM-DD Format) are required to lock workspace';
```

Comment:
the user does not control whether the date is valid because they are picking from a date widget

Reply:
The date input box is editable, i was able to enter dates like 2021-13-01 or just 2021-13

Reply:
sorry, you're right